ethereumj-core/src/main/java/org/ethereum/sharding/crypto/BLS381.java

```diff
@@ -0,0 +1,53 @@
+package org.ethereum.sharding.crypto;
+
+import java.math.BigInteger;
+
+/**
+ * Common interface for BLS381-12 implementation
+ */
+public interface BLS381 {
+
+    BI generatePrivate();
+
+    BI restorePrivate(BigInteger value);
+
+    BI restorePrivate(byte[] value);
```

Comment:
It seems that this method addresses more general case than restoring just private keys. Maybe it's better to call it `restoreScalar`?